TeamCode/src/main/java/org/firstinspires/ftc/teamcode/Robot.java

```diff
@@ -27,30 +30,55 @@
 
     private LinearOpMode linearOpMode;
     private boolean initWithOpMode = false;
-
-    public GripperState gripperState;
+    private boolean isGripperDisabled = false;
 
     enum GripperState {
-        OPEN, //OPEN might be used once limit switches are added for the open limit
+        MAX_OPEN, //MAX_OPEN might be used once limit switches are added for the open limit
         //TODO: Order and add limit switches for the outside and the appropriate code
         CLOSED,
-        PWMDISABLED,
-        MIDDLE;
+        MIDDLE
     }
 
-    public void retrieveGripperState() {
-        if (gripperState != GripperState.PWMDISABLED) {
-            if(touchSensor1.getState() == false && touchSensor2.getState() == false) {
-                gripperState = GripperState.CLOSED;
-            } /*else if (!outer limit switch) {
-                gripperState = GripperState.OPEN;
-            } */
-            else {
-                gripperState = GripperState.MIDDLE;
-            }
+    final double GRIPPER_MODIFY = 0.001;
+
+    public GripperState getGripperState() {
+
+        if (touchSensor1.getState() == false && touchSensor2.getState() == false) {
+             return GripperState.CLOSED;
+        } /*else if (!outer limit switch) {
+                gripperState = GripperState.MAX_OPEN;
+            } */ else {
+            return GripperState.MIDDLE;
+        }
+    }
+
+    public void moveGripperOpen() {
+        if(getGripperState() != MAX_OPEN && !isGripperDisabled) {
+            gripperOneServo.setPosition(gripperOneServo.getPosition() - GRIPPER_MODIFY);
+        }
+    }
+
+    public void moveGripperClosed() {
+        if (getGripperState() != CLOSED && !isGripperDisabled)  {
+            gripperOneServo.setPosition(gripperOneServo.getPosition() + GRIPPER_MODIFY);
         }
     }
 
+    public ServoImplEx getGripperOneServo() {
+        return gripperOneServo;
+    }
```

Comment:
In the rest of the code, you're avoiding accessing the gripper servo object. it's only being used here to allow the PWM to be temporarily disabled. Let's get rid of this method and replace it with something else that doesn't give other classes direct access to our servo object.

Reply:
Now there is a method gripperServoPwmDisable() for this purpose. It also fixes the performance issue.